test/src/Rect.cpp

```diff
@@ -0,0 +1,116 @@
+            CHECK(rectangle.top == 2);
+            CHECK(rectangle.width == 3);
+            CHECK(rectangle.height == 4);
+        }
+
+        SECTION("(Vector2, Vector2) constructor")
+        {
+            sf::Vector2i position(1, 2);
+            sf::Vector2i dimension(3, 4);
+            sf::IntRect rectangle(position, dimension);
+
+            CHECK(rectangle.left == 1);
+            CHECK(rectangle.top == 2);
+            CHECK(rectangle.width == 3);
+            CHECK(rectangle.height == 4);
+        }
+
+        SECTION("Conversion constructor")
+        {
+            sf::FloatRect sourcerectangle(1.0f, 2.0f, 3.0f, 4.0f);
+            sf::IntRect rectangle(sourcerectangle);
+
+            CHECK(rectangle.left == static_cast<int>(sourcerectangle.left));
+            CHECK(rectangle.top == static_cast<int>(sourcerectangle.top));
+            CHECK(rectangle.width == static_cast<int>(sourcerectangle.width));
+            CHECK(rectangle.height == static_cast<int>(sourcerectangle.height));
+        }
+    }
+
+    SECTION("Containment")
+    {
+        SECTION("contains(x, y)")
+        {
+            sf::IntRect rectangle(0, 0, 10, 10);
+
+            CHECK(rectangle.contains(0, 0) == true);
+            CHECK(rectangle.contains(10, 10) == false);
+            CHECK(rectangle.contains(15, 15) == false);
+        }
+
+        SECTION("contains(Vector2)")
+        {
+            sf::IntRect rectangle(0, 0, 10, 10);
+
+            CHECK(rectangle.contains(sf::Vector2i(0, 0)) == true);
+            CHECK(rectangle.contains(sf::Vector2i(10, 10)) == false);
+            CHECK(rectangle.contains(sf::Vector2i(15, 15)) == false);
+        }
+    }
+
+    SECTION("Intersection")
+    {
+        SECTION("intersects(Rect)")
+        {
+            sf::IntRect rectangle(0, 0, 10, 10);
+            sf::IntRect intersectingRectangle(5, 5, 10, 10);
+            sf::IntRect nonIntersectingRectangle(-5, -5, 5, 5);
+
+            CHECK(rectangle.intersects(intersectingRectangle) == true);
+            CHECK(rectangle.intersects(nonIntersectingRectangle) == false);
+        }
+
+        SECTION("intersects(Rect, Rect)")
+        {
+            sf::IntRect rectangle(0, 0, 10, 10);
+            sf::IntRect intersectingRectangle(5, 5, 10, 10);
+            sf::IntRect nonIntersectingRectangle(-5, -5, 5, 5);
+            sf::IntRect intersectionResult;
+
+            CHECK(rectangle.intersects(intersectingRectangle, intersectionResult) == true);
+            CHECK(intersectionResult.top == 5);
+            CHECK(intersectionResult.left == 5);
+            CHECK(intersectionResult.width == 5);
+            CHECK(intersectionResult.height == 5);
+
+            CHECK(rectangle.intersects(nonIntersectingRectangle, intersectionResult) == false);
+            CHECK(intersectionResult.top == 0);
+            CHECK(intersectionResult.left == 0);
+            CHECK(intersectionResult.width == 0);
+            CHECK(intersectionResult.height == 0);
```

Comment:
I agree. The empty rect behavior makes sense to me since if there is no intersection the area of the intersection would be "none" (hence an empty Rect). I will remove these checks for now until that behavior is documented or otherwise becomes codified.